src/custom/utils/switchToNetwork.ts

```diff
@@ -0,0 +1,47 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { hexStripZeros } from '@ethersproject/bytes'
+import { Web3Provider } from '@ethersproject/providers'
+import { CHAIN_INFO, SupportedChainId } from 'constants/chains'
+
+import { addNetwork } from 'utils/addNetwork'
+
+interface SwitchNetworkArguments {
+  library: Web3Provider
+  chainId?: SupportedChainId
+}
+
+// provider.request returns Promise<any>, but wallet_switchEthereumChain must return null or throw
+// see https://github.com/rekmarks/EIPs/blob/3326-create/EIPS/eip-3326.md for more info on wallet_switchEthereumChain
+export async function switchToNetwork({ library, chainId }: SwitchNetworkArguments): Promise<null | void> {
+  if (!library?.provider?.request) {
+    return
+  }
+  if (!chainId && library?.getNetwork) {
+    ;({ chainId } = await library.getNetwork())
+  }
+  const formattedChainId = hexStripZeros(BigNumber.from(chainId).toHexString())
+  try {
+    await library?.provider.request({
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: formattedChainId }],
+    })
+  } catch (error) {
+    // 4902 is the error code for attempting to switch to an unrecognized chainId
+    if (error.code === 4902 && chainId !== undefined) {
+      const info = CHAIN_INFO[chainId]
+
+      // MOD - need to handle these errors, else loops
```

Comment:
L33 - 42 (`try`/`catch`) block is the MOD. Without proper error handling this loops